src/libfetchers/git.cc

```diff
@@ -397,6 +393,9 @@ struct GitInputScheme : InputScheme
                 utimes(localRefFile.c_str(), times);
             }
 
+            // we will only use read-only operations on specific revisions in the remainder
+            cacheDirLock.unlock();
+
```

Comment:
```suggestion
```

This is unnecessary since the lock is released when `cacheDirLock` goes out of scope.

Reply:
Yeah, I mostly did it to have something explicit to attach the comment to. Moved it to the end of the block instead.